spec/views/idv/doc_auth/agreement.html.erb_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+describe 'idv/doc_auth/agreement' do
+  let(:flow_session) { {} }
+
+  before do
+    allow(view).to receive(:flow_session).and_return(flow_session)
+    allow(view).to receive(:user_signing_up?).and_return(false)
+    allow(view).to receive(:url_for).and_wrap_original do |method, *args, &block|
+      method.call(*args, &block)
+    rescue
+      ''
+    end
```

Comment:
Why is this needed? Is there a way to test the code change more directly by making assertions about what FakeAnalytics receives?

Reply:
In this case it's just a view test, and the redirect / analytics code already has tests elsewhere. What I want to test here is that the view contains the link I think it should contain.

These step views use `url_for` in wiring up their `SimpleForms`. This ends up throwing `No route matches {:action=>"agreement", :controller=>"idv/doc_auth"}` during the view spec -- I'm not sure why (I assume FSM???).

The existing spec for the welcome view [gets around this](https://github.com/18F/identity-idp/blob/main/spec/views/idv/doc_auth/welcome.html.erb_spec.rb#L15) by mocking `url_for` to always return `"https://www.example.com"`. However, I'm trying to test code that uses the `policy_redirect_url` helper, which looks like it uses `url_for` under the hood, so I need that code to actually return the original, but I don't really care about the `url_for` call used by `SimpleForm`.

Agree this is weird, and would love to know why `url_for` is throwing but I think I might be out of ideas.

Reply:
I'm willing to dig into this with you, and also willing to trust that you landed on a good answer in the circumstances. Let's revisit post-FSM!